python/paddle/fluid/dygraph/dygraph_to_static/utils.py

```diff
@@ -619,3 +634,308 @@ def _is_node_with_tensor(self, node, name_id):
 
     def get_compare_nodes_with_tensor(self):
         return self._compare_node_tenor_set
+
+
+class NameNodeReplaceTransformer(gast.NodeTransformer):
+    """
+    This class transform specfice gast.Name node to replace node
+    """
+
+    def __init__(self, root_node, target_name, replace_node):
+        assert isinstance(target_name, str)
+        self.target_name = target_name
+        self.replace_node = replace_node
+
+        self.visit(root_node)
+
+    def visit_Name(self, node):
+        if node.id == self.target_name:
+            return self.replace_node
+        return node
+
+
+class ForNodeParser(object):
```

Comment:
Since python call this kind of classes Visitor, should we just call ForNodeVisitor to avoid adding new concept?

Reply:
thx, I will fix this problem in next PR